canonical/cn.ubuntu.com
# Fix blog newsletter form

## Done

Fix /blog newsletter signup form by removing legacy code (form-validation.js)

I believe this is legacy code because:
- Most of the `.whatever` classes used by this script are nowhere to be found.
- It has code that adds URL for things like whitepapers, but whitepaper asset URLs are now handled by Discourse module's engage pages, which are provided by the respective discourse topic.
- It has a lot of code just to submit the newsletter subscription, on u.com we don't even use JS.
- We don't seem to have interactive forms (modals) on cn.u.com for the "get in touch" buttons

## QA

### Test that https://cn-ubuntu-com-674.demos.haus/blog newsletter works
- Go to https://cn-ubuntu-com-674.demos.haus/blog, fill up the form, and check that it submits successfully now. Previously, it was failing with an error in console
- Check the success messages shows up on top

### Test that engage page forms work
- Go to https://cn-ubuntu-com-674.demos.haus/engage pick an engage page (with a form)
- Fill up the form, submit, it should redirect you to the thank-you page as usual, with the download button (green)

### Test contact forms

- Go to https://cn-ubuntu-com-674.demos.haus/contact
- Fill up the form and submit, it should redirect to the usual thank-you page


Any other form you can think of.

## Issue / Card

Fixes https://github.com/canonical/cn.ubuntu.com/issues/673
